server/src/main/java/org/opensearch/indices/recovery/RemoteRecoveryTargetHandler.java

```diff
@@ -129,7 +132,13 @@ public void prepareForTranslogOperations(int totalTranslogOps, ActionListener<Vo
         );
         final Writeable.Reader<TransportResponse.Empty> reader = in -> TransportResponse.Empty.INSTANCE;
         final ActionListener<TransportResponse.Empty> responseListener = ActionListener.map(listener, r -> null);
-        retryableTransportClient.executeRetryableAction(action, request, responseListener, reader);
+        if (remoteStoreEnabled) {
+            // If remote store is enabled, during the prepare_translog phase, translog is also downloaded on the
+            // target host along with incremental segments download. This
+            retryableTransportClient.executeRetryableAction(action, request, translogOpsRequestOptions, responseListener, reader);
+        } else {
+            retryableTransportClient.executeRetryableAction(action, request, responseListener, reader);
+        }
```

Comment:
Fortunately both long(30m) and default timeouts(15m) are dynamic in nature.

Reply:
Reusing the `internalActionLongTimeout` (defaults to 30 mins) which is already being used in `internal:index/shard/recovery/translog_ops` transport action (applicable for docrep). In the existing state, it was using `indices.recovery.internal_action_retry_timeout` which defaults to 1m and hence if you see [here](https://github.com/opensearch-project/OpenSearch/issues/10727#issue-1951058192), we can see `Caused by: ReceiveTimeoutTransportException[[30ddfff069f4928ae20b3f7f9046e41f][][internal:index/shard/recovery/prepare_translog] request_id [1304756] timed out after [60018ms]]
`.

Reply:
Can we use the same `internalActionLongTimeout ` for both remote and non-remote store. Would have ideally preferred it.